app/src/pages/[resourceType]/[id].tsx

```diff
@@ -27,14 +33,49 @@ export default function ResourceIDPage({ jsonData }: InferGetServerSidePropsType
     return encodedCql ? Buffer.from(encodedCql, 'base64').toString() : null;
   }, [jsonData]);
 
+  const ctx = trpc.useContext();
+
+  const draftMutation = trpc.draft.createDraft.useMutation({
+    onSuccess: data => {
+      notifications.show({
+        title: `${jsonData.resourceType} Cloned!`,
+        message: `${jsonData.resourceType} successfully cloned`,
+        icon: <CircleCheck />,
+        color: 'green'
+      });
+      router.push(`/authoring/${jsonData.resourceType}/${data.draftId}`);
+      ctx.draft.getDraftCounts.invalidate();
+    },
+    onError: e => {
+      notifications.show({
+        title: `${jsonData.resourceType} Clone Failed!`,
+        message: `Attempt to clone ${jsonData.resourceType} failed with message: ${e.message}`,
+        icon: <AlertCircle />,
+        color: 'red'
+      });
+    }
+  });
+
+  const router = useRouter();
+
+  const cloneResource = () => {
+    const newClonedResource = jsonData;
+    newClonedResource.id = uuidv4();
+    newClonedResource.status = 'draft';
+    draftMutation.mutate({ resourceType, draft: newClonedResource });
+  };
```

Comment:
I'm not sure if this what is happening, but would it be possible to only get the `cloneResource` after the `clone to draft artifact button` is clicked and is successful?

Reply:
Do you mean like only see the JSON of the resource without those input buttons? Or something else?

Reply:
Sorry, I was a bit confused on this piece of code earlier and I think it led me to mistakenly think something was wrong. I'm still a bit confused but everything looks good to me !